### ISSUE TITLE
"Compute CTC targets failed!"

### ISSUE THREAD

**Issue by stefanCCS** [tesseract-ocr/tesstrain#210]
Hi,
I get the following error message during training. The training goes on, and the result looks ok, but still it might be worth to have a look at this:
```
Compute CTC targets failed!
````
The training I have started with:
```
make MAX_ITERATIONS=300 MODEL_NAME=swerror02 GROUND_TRUTH_DIR=~/tesstrain/data/swerror02-ground-truth/ START_MODEL=grc TESSDATA=/usr/share/tesseract-ocr/5/tessdata training
```
One topic I changed in comparison to standard installation:
I have changed shuffle.py, so, that the files are not shuffeled at all (to make any reproducing of failures a bit easier) like this:
```
# Then shuffle the lines.
if len(sys.argv) > 1:
        if (sys.argv[1] != "0"):
                random.shuffle(lines)
```
The ground-truth-data you can find here attached:
[swerror02-ground-truth.zip](https://github.com/tesseract-ocr/tesstrain/files/5673993/swerror02-ground-truth.zip)

**Comment by Shreeshrii** [comment]
The total groundtruth images are very limited. Out of that many images are cropped incorrectly. In some cases the bottom part of one text line and top of next text line are in the image. In other cases 4-5 text lines are in the image and are cropped in multiple pieces.

Additionally some lines are at very large font size. For finetune with such small training set all lines need to be of similar type.

In thousands of lines a few such samples may not effect much, but when almost 15 out of 60 lines have a problem, you will not get good result.

Some problem images posted below.

![GT-000009](https://user-images.githubusercontent.com/5095331/101867835-1bad1200-3ba2-11eb-943a-e3026cd96c1f.png)
![GT-000004](https://user-images.githubusercontent.com/5095331/101867840-1d76d580-3ba2-11eb-80da-0cbbcefca826.png)
![GT-000005](https://user-images.githubusercontent.com/5095331/101867841-1d76d580-3ba2-11eb-9cd7-7f70d47218e4.png)
![GT-000006](https://user-images.githubusercontent.com/5095331/101867843-1e0f6c00-3ba2-11eb-9e74-2f9f04ba1866.png)
![GT-000007](https://user-images.githubusercontent.com/5095331/101867844-1f409900-3ba2-11eb-9e67-144c9bc8dc02.png)
![GT-000008](https://user-images.githubusercontent.com/5095331/101867847-1fd92f80-3ba2-11eb-8292-d641accdd303.png)
![GT-000046](https://user-images.githubusercontent.com/5095331/101867903-472ffc80-3ba2-11eb-98c2-3ac8d8e87023.png)
![GT-000047](https://user-images.githubusercontent.com/5095331/101867908-48612980-3ba2-11eb-9ffc-c4791f9e805f.png)
![GT-000048](https://user-images.githubusercontent.com/5095331/101867910-48f9c000-3ba2-11eb-8948-08abb8f53797.png)
![GT-000049](https://user-images.githubusercontent.com/5095331/101867912-49925680-3ba2-11eb-8d75-58687676840d.png)
![GT-000043](https://user-images.githubusercontent.com/5095331/101867913-49925680-3ba2-11eb-9431-f75c7b5bd59f.png)
![GT-000044](https://user-images.githubusercontent.com/5095331/101867914-4a2aed00-3ba2-11eb-8ed9-660108f572a6.png)
![GT-000045](https://user-images.githubusercontent.com/5095331/101867917-4ac38380-3ba2-11eb-8e21-bb11ae69028b.png)

**Comment by stefanCCS** [comment]
Many thanks for providing this fast answer.
For explanation: I am aware of that my examle I have provided here is a very small ground truth. In the real project my ground truth is about 30000 lines.
I have put this here to understand, why I get this "Compute CTC targets failed" error.
And for this I am still not sure, what is the root cause for this:
- It should be NOT the small ground truth number of files, as I get the same errors for my whole ground truth
- Maybe it is related to the badly cropped lines?
- Or something else?

And, what I am also interested in: Is this error "Compute CTC targets failed" somehow critical for my training?

Please, give me another advice to these questions.

**Comment by Shreeshrii** [comment]
The problem is with badly cropped lines. If I remove them CTC errors go away. See attached log.

[swerror02.log](https://github.com/tesseract-ocr/tesstrain/files/5678361/swerror02.log)

>Is this error "Compute CTC targets failed" somehow critical for my training?

It depends on how many and what proportion of total training data they are? You could ignore if it was 15 out of all 30000 lines, you shouldn't if it is 15 out of 60 lines.

**Comment by stefanCCS** [comment]
Again, many thanks for answering.
This means I will take care mainly of the not good input data:
Concerning "CTC" error - I will take care of the badly cropped lines.
And, of course I will also take care of the other bad input data.
